### PR TITLE
A header based on 2 asset images

### PR DESCRIPTION
**Pull request by jakehschwartz**
I was also interested in the header functionality, so I made an attempt at doing it with 2 image assets. I don't know what interface you were thinking about, but this is probably what my app would need. Im happy to add more if you had addition ideas!

### CODE REVIEW

**Comment by jksevend** [comment]
Hello Jake,

I have created a trello board to keep track of further changes. Do you have an account? I would invite you via ur name or email. Also we shouldn't push code directly to the main branch. Please create a branch called 'SN-X' and publish your code to it :) 

I really like your way of putting images in the header. I was thinking about a more abstract approach by defining widgets to put in the header to give the user more freedom. I'm currently working on that but still have some problems figuring out providable onTap callbacks. Tomorrow I should be able to release the new branch 💯 

Then you can take a look at it and see if you like it.

Check this image for an example:
header

File: lib/api/side_navigation_bar.dart [hunk 1]
```diff
@@ -90,7 +96,7 @@ class _SideNavigationBarState extends State<SideNavigationBar> with SingleTicker
             child: Column(
               children: [
                 // Header
-                // TODO: implement header
+                if (widget.header != null) widget.header!.generateWidget(context, expanded, width),
```
Comment (jksevend) [inline lib/api/side_navigation_bar.dart:99]: I was thinking about making the header required. What do you think about that?

**Comment by jakehschwartz** [comment]
Yeah I think I have a Trello account. My email is jakehschwartz@gmail.com 

In terms of branches, I dont have the ability to make branches in your project, so that's why my PR is pointing to main.


I like your idea of something more generic. Maybe the header can be a custom interface that they have to inherit that has a collapse mode and an expanded mode?

File: lib/api/side_navigation_bar.dart (hunk 1 above)
Comment (jakehschwartz) [reply lib/api/side_navigation_bar.dart:99]: Its your project :)

Seemed easy enough to make it optional, that was the only reason I did it.

**Comment by jksevend** [comment]
I just invited you. Ohh whooopsie.. sorry I am pretty new to this github stuff :) How can I give you the privileges to push new branches ?

Yes exactly like that! 👍I plan to add a custom collapsed and not collapsed `Widget` to the header.

**Comment by jakehschwartz** [comment]
There are a few ways. You can make me a collaborator, which you did, but you don't want to do that for everyone.

The way I have done it in my few projects is just have people merge to master and then have the releases instead be based on the tags/releases functionality in github

**Pull request closed without merging**